docs/src/aws-construct-lib.rst

```diff
@@ -103,9 +103,274 @@ were part of your app.
+definition of the underlying resources are included in the resulting template.
+
+In the fullness of time, the APIs provided by AWS constructs are expected to
+support all the services and capabilities offered by AWS, but we are aware that
+the library still has many gaps both at the service level (some services simply
+don't have any constructs yet) and at the resource level (an AWS construct
+exists, but some features are missing).
+
+.. note::
+
+   If you encounter a missing capability in the AWS Construct Library, whether
+   it is an entire library, a specific resource or a feature, please consider to
+   `raise an issue <https://github.com/awslabs/aws-cdk/issues/new>`_ on GitHub,
+   and letting us know.
+
+This topic covers the following use cases:
+
+- How to access CloudFormation the internal resources encapsulated by an AWS construct
+- How to specify resource options such as metadata, dependencies on resources
+- How to add overrides to a CloudFormation resource and property definitions
+- How to directly define low-level CloudFormation resources without an AWS construct
+
+You can also find more information on how to work directly with the AWS
+CloudFormation layer under :py:doc:`cloudformation`.
+
+Accessing Low-level Resources
+-----------------------------
+
+You can use :py:meth:`construct.findChild(id) <@aws-cdk/cdk.Construct.findChild>`
+to access any child of this construct by it's construct ID. By convention, the "main"
+resource of any AWS Construct will always be called ``"Resource"``.
+
+The following example shows how to access the underlying S3 bucket resource
+given an :py:class:`s3.Bucket <@aws-cdk/s3.Bucket>` construct:
+
+.. code-block:: ts
+
+   // let's create an AWS bucket construct
+   const bucket = new s3.Bucket(this, 'MyBucket');
+
+   // we use our "knoweledge" that the main construct is called "Resource" and
+   // that it's actual type is s3.cloudformation.BucketResource; const
+   bucketResource = bucket.findResource('Resource') as s3.cloudformation.BucketResource;
+
+At this point, ``bucketResource`` represents the low-level CloudFormation resource of type
+:py:class:`s3.cloudformation.BucketResource <@aws-cdk/aws-s3.cloudformation.BucketResource>`
+encapsulated by our bucket.
+
+:py:meth:`construct.findChild(id) <@aws-cdk/cdk.Construct.findChild>` will fail
+if the child could not be located, which means that if the underlying L2 changes
+the IDs or structure for some reason, synthesis will fail.
+
+It is also possible to use :py:meth:`construct.children <@aws-cdk/cdk.Construct.children>` for more
+advanced queries. For example, we can look for a child that has a certain CloudFormation resource
+type:
+
+.. code-block:: ts
+
+   const bucketResource =
+      bucket.children.find(c => (c as cdk.Resource).resourceType === 'AWS::S3::Bucket')
+      as s3.cloudformation.BucketResource;
+
+From that point, users are interacting with the strong-typed L1 resources (which
```

Comment:
Don't know that the word "strong-typed" adds a lot here.

Reply:
👍